src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -84,16 +85,39 @@ protected function setDefinition($id, Definition $definition)
         }
     }
 
-    private function findClasses($namespace, $pattern)
+    private function findClasses($namespace, $pattern, $excludePattern)
     {
         $parameterBag = $this->container->getParameterBag();
+
+        $excludePaths = array();
+        $excludePrefix = null;
+        if ($excludePattern) {
+            $excludePattern = $parameterBag->unescapeValue($parameterBag->resolveValue($excludePattern));
+            foreach ($this->glob($excludePattern, true, $resource) as $path => $info) {
+                if (null === $excludePrefix) {
+                    $excludePrefix = $resource->getPrefix();
+                }
+
+                // realpath to make sure Windows slashes are consistent
+                $excludePaths[realpath($path)] = true;
```

Comment:
`realpath()` with return `false` if the $path does not exist. You should take care of that. We try to avoid using `realpath()` in Symfony (another issue is when you are using a phar -- not sure we need to support that case anyway).